### ISSUE TITLE
Default locale & browser language detection confusion

### ISSUE THREAD

**Issue by paulgv** [nuxt-community/nuxt-i18n#114]
### Version

[v5.0.3](https://github.com/nuxt-i18n/releases/tag/v5.0.3)

### Reproduction link

[https://github.com/nuxt-community/nuxt-i18n](https://github.com/nuxt-community/nuxt-i18n)

### Steps to reproduce

Quoting @mapeal2 
https://cmty.app/nuxt/nuxt-i18n/issues/c96

```
Given 2 locales ('en', 'fr'), default locale is 'en' and the preferred browser language is 'fr', with the 'prefixexceptdefault' option, the steps to reproduce it would be:
Visit home page '/' --> browser language is detected and redirected to '/fr', which is correct.
Close the browser.
Open a new browser and enter the home page ('/') again 
```

### What is expected ?

```
I would expect that, when opening the home page again, the application would be redirected to '/fr' instead of staying in '/'.

```

### What is actually happening?

```
 'en' locale is chosen, ignoring cookie.
```

<div align="right"><sub><em>This bug report is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt-i18n/issues/c116">#c116</a>)</em></sub></div>

**Comment by Aperrix** [comment]
Same bug with v5.4.4 of nuxt-i18n (using the [nuxt-i18n rxample](https://github.com/paulgv/nuxt-i18n-example) by @paulgv 

```javascript
modules: ['nuxt-i18n' {
  locales: [
    { code: 'fr', iso: 'fr-FR', name: 'Français' },
    { code: 'en', iso: 'en-GB', name: 'English' },
    { code: 'de', iso: 'de-DE', name: 'Deutsch' },
    { code: 'es', iso: 'es-ES', name: 'Español' },
    { code: 'it', iso: 'it-IT', name: 'Italiano' },
    { code: 'pt', iso: 'pt-PT', name: 'Português' }
  ],
  defaultLocale: 'en',
  detectBrowserLanguage: {
    useCookie: true,
    alwaysRedirect: true
  },
  vueI18n: {
    messages: { fr, en, de, es, it, pt }
  }
}],
```
![image](https://user-images.githubusercontent.com/26674548/48317506-a56d0880-e5f3-11e8-873b-6ae896dd882f.png)

**Comment by pschaub** [comment]
@paulgv the user @rchl has a fix for this. I have not tested it yet. See https://github.com/rchl/nuxt-i18n/commits/fix/always-redirect

But first of all we need to merge https://github.com/nuxt-community/nuxt-i18n/pull/266 because it is based on a possible fix for this issue

**Comment by paulgv** [comment]
@pschaub Thanks for the heads up! @rchl would you care opening a PR with your work around cookies? I'll make sure to review it ASAP.

**Comment by rchl** [comment]
@paulgv @pschaub created #283 with the fix. Basically fixed so that with `detectBrowserLanguage.useCookie` and `detectBrowserLanguage.alwaysRedirect` enabled, redirection will always happen (whether navigating server side or on the client).

I did wonder though, while making the fix, what is really the point of using `detectBrowserLanguage.useCookie = true` and `detectBrowserLanguage.alwaysRedirect = false`. I don't know why would someone want that behavior (to redirect only on first visit).

**Comment by paulgv** [comment]
@rchl thanks! I'll have a look asap.

**Comment by rchl** [comment]
Fix is integrated. This can be resolved.

**Comment by paulgv** [comment]
Fixed in #283
